tests/Unit/Evolution/Systems/Cce/Test_KleinGordonWorldtubeData.cpp

```diff
@@ -0,0 +1,362 @@
+          ? KleinGordonWorldtubeH5BufferUpdater{filename}
+          : KleinGordonWorldtubeH5BufferUpdater{filename, extraction_radius};
+
+  size_t time_span_start = 0;
+  size_t time_span_end = 0;
+  Variables<klein_gordon_input_tags> coefficients_buffers_from_file{
+      (buffer_size + 2 * interpolator_length) * square(computation_l_max + 1)};
+  buffer_updater.update_buffers_for_time(
+      make_not_null(&coefficients_buffers_from_file),
+      make_not_null(&time_span_start), make_not_null(&time_span_end),
+      target_time, computation_l_max, interpolator_length, buffer_size);
+
+  const auto& time_buffer = buffer_updater.get_time_buffer();
+  for (size_t i = 0; i < time_buffer.size(); ++i) {
+    CHECK(time_buffer[i] == approx(target_time - 0.1 + 0.01 * i));
+  }
+  CHECK(buffer_updater.get_extraction_radius() == 100.0);
+
+  // Test the `pup` function of `KleinGordonWorldtubeH5BufferUpdater`.
+  // Serialize and deserialize `buffer_updater` and repeat the checks above.
+  auto serialized_and_deserialized_updater =
+      serialize_and_deserialize(buffer_updater);
+
+  Variables<klein_gordon_input_tags> coefficients_buffers_from_serialized{
+      (buffer_size + 2 * interpolator_length) * square(computation_l_max + 1)};
+  size_t time_span_start_from_serialized = 0;
+  size_t time_span_end_from_serialized = 0;
+  serialized_and_deserialized_updater.update_buffers_for_time(
+      make_not_null(&coefficients_buffers_from_serialized),
+      make_not_null(&time_span_start_from_serialized),
+      make_not_null(&time_span_end_from_serialized), target_time,
+      computation_l_max, interpolator_length, buffer_size);
+
+  const auto& time_buffer_from_serialized =
+      serialized_and_deserialized_updater.get_time_buffer();
+  for (size_t i = 0; i < time_buffer.size(); ++i) {
+    CHECK(time_buffer_from_serialized[i] ==
+          approx(target_time - 0.1 + 0.01 * i));
+  }
+  CHECK(serialized_and_deserialized_updater.get_extraction_radius() == 100.0);
+
+  // Compare the modal data of `buffer_updater` and
+  // `serialized_and_deserialized_updater`, which goes through the write-read
+  // process, with what we generated earilier. The fiducial modal data are
+  // stored in an object of `KleinGordonDummyBufferUpdater`.
+  time_span_start = 0;
+  time_span_end = 0;
+  const KleinGordonDummyBufferUpdater dummy_buffer_updater{
+      time_buffer, extraction_radius, amplitude, frequency, computation_l_max};
+  Variables<klein_gordon_input_tags> expected_coefficients_buffers{
+      (buffer_size + 2 * interpolator_length) * square(computation_l_max + 1)};
+  dummy_buffer_updater.update_buffers_for_time(
+      make_not_null(&expected_coefficients_buffers),
+      make_not_null(&time_span_start), make_not_null(&time_span_end),
+      target_time, computation_l_max, interpolator_length, buffer_size);
+
+  Approx modal_approx =
+      Approx::custom()
+          .epsilon(std::numeric_limits<double>::epsilon() * 1.0e5)
+          .scale(1.0);
```

Comment:
Is the written data really this different from the read in data? I would not expect that.

Reply:
Sorry, that number was a relic from `Test_WorldtubeData.cpp`. The difference in my case is Xe-14, so I need 1e-13 to let the tests pass, which is roughly `1000 * epsilon`. I've updated the number.